Sources/PostgresNIO/Connection/PostgresConnection+Notifications.swift

```diff
@@ -0,0 +1,56 @@
+import NIO
+
+public final class PostgresListenContext {
+    var stopper: (() -> Void)?
+
+    public func stop() {
+        stopper?()
+        stopper = nil
+    }
+}
+
+extension PostgresConnection {
+    @discardableResult
+    public func listen(channel: String, handler notificationHandler: @escaping (PostgresMessage.NotificationResponse, PostgresListenContext) -> Void) -> PostgresListenContext {
+        let listenContext = PostgresListenContext()
+        let channelHandler = PostgresNotificationHandler(channel: channel, notificationHandler: notificationHandler, listenContext: listenContext)
+        let pipeline = self.channel.pipeline
+        _ = pipeline.addHandler(channelHandler, name: nil, position: .before(requestHandler))
+        listenContext.stopper = { [pipeline, unowned channelHandler] in
+            _ = pipeline.removeHandler(channelHandler)
+        }
+        return listenContext
+    }
+}
+
+final class PostgresNotificationHandler: ChannelInboundHandler, RemovableChannelHandler {
+    typealias InboundIn = PostgresMessage
+    typealias InboundOut = PostgresMessage
+
+    let channel: String
+    let notificationHandler: (PostgresMessage.NotificationResponse, PostgresListenContext) -> Void
```

Comment:
NIO methods tend to pass `context` first. It might be nice to follow that pattern.